scripts/update_password.php

```diff
@@ -40,11 +61,57 @@
 }
 
 $auth = new Hm_Auth_DB($config);
-if ($user && $pass) {
-    if ($auth->change_pass($user, $pass)) {
-        die("Password Updated\n\n");
+
+if (! $auth->check_credentials($user, $old_pass)) {
```

Comment:
The check here will fail when user didn't supply an old password - it needs to be in the block that handles conversion using the old password.